### PR TITLE
first branch

### PR DESCRIPTION
**Pull request by raghavgarg098**

### CODE REVIEW

File: src/com/tsa/Main.java [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 public class Main {
 
-  public static void main(String[] args) {
+  public static void main(String[] args)
```
Comment (ankitg364) [inline src/com/tsa/Main.java:5]: Revert this change

File: src/com/tsa/TweetManager.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.tsa;
+
+public class TweetManager
+{
+    public static arraylist<string> gettweets(string topic)
```
Comment (ankitg364) [inline src/com/tsa/TweetManager.java:5]: getTweets. Method names should be camel Case. Also is string a valid object in java ? Shouldn't it be String

File: src/com/tsa/TweetsManage.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import twitter4j.Query;
+import twitter4j.QueryResult;
+import twitter4j.Status;
+import twitter4j.Twitter;
+import twitter4j.TwitterException;
+import twitter4j.TwitterFactory;
+
+public class TweetsManage {
+
+    public static ArrayList<String> getTweets(String topic)
```
Comment (ankitg364) [inline src/com/tsa/TweetsManage.java:14]: Where are you calling this method

File: src/com/tsa/TweetsManage.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        try {
+            Query query = new Query(topic);
+            QueryResult result;
+            do {
+                result = twitter.search(query);
+                List<Status> tweets = result.getTweets();
+                for (Status tweet : tweets) {
+                    tweetList.add(tweet.getText());
+                }
+            } while ((query = result.nextQuery()) != null);
```
Comment (ankitg364) [inline src/com/tsa/TweetsManage.java:27]: Maybe rate limiting error happens when you call search again and again. You could stop the call once you have fetched some number of tweets like 1000.

File: src/com/tsa/TweetsManage.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import java.util.List;
+
+import twitter4j.Query;
+import twitter4j.QueryResult;
+import twitter4j.Status;
+import twitter4j.Twitter;
+import twitter4j.TwitterException;
+import twitter4j.TwitterFactory;
+
+public class TweetsManage {
```
Comment (ankitg364) [inline src/com/tsa/TweetsManage.java:12]: We are not managing any tweets. TweetsFetcher is a better name. Also names of the class should be a noun not a verb.

File: src/com/tsa/TweetsManage.java (hunk 3 above)
Comment (raghavgarg098) [reply src/com/tsa/TweetsManage.java:14]: i am not calling anywhere now I just created a method for now

File: src/com/tsa/TweetsManage.java (hunk 5 above)
Comment (raghavgarg098) [reply src/com/tsa/TweetsManage.java:12]: ok

File: src/com/tsa/TweetManager.java (hunk 2 above)
Comment (raghavgarg098) [reply src/com/tsa/TweetManager.java:5]: ok

File: src/com/tsa/Main.java (hunk 1 above)
Comment (raghavgarg098) [reply src/com/tsa/Main.java:5]: can't understand this change

Comment (ankitg364) [reply src/com/tsa/Main.java:5]: You can revert changes by comparing with branch master in intellij.

File: src/MyPropFile.properties [hunk 6]
```diff
@@ -0,0 +1 @@
+annotators = tokenize, ssplit, parse, sentiment
```
Comment (ankitg364) [inline src/MyPropFile.properties:1]: What's this file? Is this needed ?

File: src/com/tsa/TweetsFetcher.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+import twitter4j.Twitter;
+import twitter4j.TwitterException;
+import twitter4j.TwitterFactory;
+
+public class TweetsFetcher {
+
+    public static void main (String[] args)
+    {
+        Twitter twitter = new TwitterFactory().getInstance();
+        String topic="Modi";
```
Comment (ankitg364) [inline src/com/tsa/TweetsFetcher.java:16]: Take this as input from the user

File: src/com/tsa/TweetsFetcher.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+        int i=0;
```
Comment (ankitg364) [inline src/com/tsa/TweetsFetcher.java:17]: Rename to tweetsCount

File: src/com/tsa/TweetsFetcher.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+            Query query = new Query(topic);
+            QueryResult result;
+            do {
+                result = twitter.search(query);
+                List<Status> tweets = result.getTweets();
+                for (Status tweet : tweets) {
+
+                    System.out.println(tweet.getText()+" ");
+                    i++;
+                    if(i==100)
```
Comment (ankitg364) [inline src/com/tsa/TweetsFetcher.java:28]: No need for this. Once we have fetched the tweets. We can add to the list.

Comment (ankitg364) [inline src/com/tsa/TweetsFetcher.java:28]: Declare this as private static final constant in the class

File: src/com/tsa/TweetsFetcher.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+import twitter4j.Query;
+import twitter4j.QueryResult;
+import twitter4j.Status;
+import twitter4j.Twitter;
+import twitter4j.TwitterException;
+import twitter4j.TwitterFactory;
+
+public class TweetsFetcher {
+
+    public static void main (String[] args)
```
Comment (ankitg364) [inline src/com/tsa/TweetsFetcher.java:13]: Just have fetch API in this class with take in String Object and returns a List<String> of tweets. Also how does it work if you need to search multiple words. Add a javadoc to the method to explain this

File: src/MyPropFile.properties (hunk 6 above)
Comment (raghavgarg098) [reply src/MyPropFile.properties:1]: not now but later

File: src/com/tsa/TweetsFetcher.java (hunk 10 above)
Comment (raghavgarg098) [reply src/com/tsa/TweetsFetcher.java:13]: resolved

File: src/com/tsa/TweetsFetcher.java (hunk 9 above)
Comment (raghavgarg098) [reply src/com/tsa/TweetsFetcher.java:28]: resolved

Comment (raghavgarg098) [reply src/com/tsa/TweetsFetcher.java:28]: resolved

File: src/com/tsa/TweetsFetcher.java (hunk 8 above)
Comment (raghavgarg098) [reply src/com/tsa/TweetsFetcher.java:17]: library functions

File: src/com/tsa/TweetsFetcher.java (hunk 7 above)
Comment (raghavgarg098) [reply src/com/tsa/TweetsFetcher.java:16]: in the main function this was for test

File: src/com/tsa/TweetsFetcher.java [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+import twitter4j.Twitter;
+import twitter4j.TwitterException;
+import twitter4j.TwitterFactory;
+
+public class TweetsFetcher {
+
+    public static ArrayList<String> fetchtweets(String topic) {
+        Twitter twitter = new TwitterFactory().getInstance();
+        ArrayList<String> tweetList = new ArrayList<String>();
+        private static final constant i = 0;
```
Comment (ankitg364) [inline src/com/tsa/TweetsFetcher.java:18]: Why is it a constant?
Variables in a method are not declared private/public. These constructs are for class variables only. Also final variables once assigned cannot be updated again. 

This should just be a simple integer tweetsCounter.

File: src/com/tsa/TweetsFetcher.java [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+import twitter4j.Query;
+import twitter4j.QueryResult;
+import twitter4j.Status;
+import twitter4j.Twitter;
+import twitter4j.TwitterException;
+import twitter4j.TwitterFactory;
+
+public class TweetsFetcher {
+
+    public static ArrayList<String> fetchtweets(String topic) {
```
Comment (ankitg364) [inline src/com/tsa/TweetsFetcher.java:15]: Please add a javadoc explaining this API. Also add that it returns a maximum of 100 tweets.

Comment (raghavgarg098) [reply src/com/tsa/TweetsFetcher.java:15]: what is meant by javadoc comments?

File: src/com/tsa/TweetsFetcher.java (hunk 11 above)
Comment (raghavgarg098) [reply src/com/tsa/TweetsFetcher.java:18]: resolved
